app/services/coronavirus_pages/draft_updater.rb

```diff
@@ -0,0 +1,38 @@
+module CoronavirusPages
+  class DraftUpdater
+    DraftUpdaterError = Class.new(StandardError)
+
+    attr_reader :coronavirus_page
+
+    def initialize(coronavirus_page)
+      @coronavirus_page = coronavirus_page
+    end
+
+    delegate :content_id, :base_path, to: :coronavirus_page
+
+    def content_builder
+      @content_builder ||= CoronavirusPages::DetailsBuilder.new(coronavirus_page)
+    end
+
+    def payload
+      if content_builder.success?
+        CoronavirusPagePresenter.new(content_builder.data, base_path)
+      else
+        raise DraftUpdaterError, content_builder.errors.to_sentence
+      end
+    end
+
+    def send
+      @send ||= Services.publishing_api.put_content(content_id, payload)
```

Comment:
Might we want this to run multiple times? Possibly not, but it might be a little surprising to have a one-shot updater? (Arguing with myself now)

Reply:
I think having a long lived object handling multiple requests is bad OOP practice. If you want to avoid mutation and threading issues the best policy is to have an instance of an object handle each request. So each instance of the draft updater handles one update process. If you want to run the update again, you initiate another instance.